pkg/occlient/occlient.go

```diff
@@ -425,18 +541,23 @@ func getAppRootVolumeName(dcName string) string {
 // inputPorts is the array containing the string port values
 func (c *Client) NewAppS2I(name string, builderImage string, gitUrl string, labels map[string]string, annotations map[string]string, inputPorts []string) error {
 
-	imageName, imageTag, _, err := ParseImageName(builderImage)
+	imageNS, imageName, imageTag, _, err := ParseImageName(builderImage)
 	if err != nil {
 		return errors.Wrap(err, "unable to parse image name")
 	}
 
 	var containerPorts []corev1.ContainerPort
 	if len(inputPorts) == 0 {
-		containerPorts, err = c.GetExposedPorts(imageName, imageTag)
+		imageNS, containerPorts, err = c.GetExposedPorts(imageNS, imageName, imageTag)
```

Comment:
Yes because, only in case of user chosen imagestream, it'll be non-empty from ParseImageName, else if user has not chosen Namespace, it needs to be resolved by GetImageStream function.
In case namespace is explicitly passed also, GetImageStream returns the passed namespace itself so, the overwrite is safe